drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -1149,7 +1177,11 @@ static bool rswitch_rx_chain(struct net_device *ndev, int *quota, struct rswitch
 			skb_reset_network_header(skb);
 
 			ethhdr = (struct ethhdr*)skb_mac_header(skb);
-			skb_set_network_header(skb, sizeof(*ethhdr));
+
+			if (skb_is_vlan(skb))
+				skb_set_network_header(skb, sizeof(*ethhdr) + 4);
```

Comment:
Nit: Maybe it's better to define something like: 
#define VLAN_HEADER_OFFSET (4)
and avoid using magic numbers?